test/Concurrency/custom_executor_enqueue_deprecation_on_executor_extension.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: %target-typecheck-verify-swift -disable-availability-checking
+// REQUIRES: concurrency
+
+// rdar://106849189 move-only types should be supported in freestanding mode
+// UNSUPPORTED: freestanding
+
+
+extension Executor {
+  func enqueue(_ job: UnownedJob) { // expected-warning{{'Executor.enqueue(UnownedJob)' is deprecated as a protocol requirement; conform type 'NoneExecutor' to 'Executor' by implementing 'func enqueue(ExecutorJob)' instead}}
```

Comment:
Now that we prevent warnings about the "default implementations in the stdlib" on purpose by looking for it, rather than relying on the LoC check, we add this test to make sure we've not accidentally prevented those warnings we DO want to emit -- whenever anything other than the Concurrency module does implement these.